bigquery_etl/query_scheduling/task.py

```diff
@@ -418,6 +431,25 @@ def _duplicate_dependency(task):
                             dependencies.append(task)
                         break  # stop after the first match
 
+        if (
+            self.date_partition_parameter is not None
+            and self.date_partition_offset is None
+        ):
+            # adjust submission_date parameter based on whether upstream tasks have
+            # date partition offsets
+            date_partition_offsets = [
+                dependency.date_partition_offset
+                for dependency in dependencies
+                if dependency.date_partition_offset
+            ]
+
+            if len(date_partition_offsets) > 0:
+                self.date_partition_offset = (
+                    min(date_partition_offsets)
+                    if self.date_partition_offset is None
+                    else self.date_partition_offset
+                )
```

Comment:
Since `self.date_partition_offset` is checked above, this can be simplified.
```suggestion
                self.date_partition_offset = min(date_partition_offsets)
```